htmd/smallmol/smallmol.py

```diff
@@ -1,105 +1,654 @@
+        self.__dict__['element'] = np.array(elements)
+        self.__dict__['chiral'] = np.array(chirals)
+        self.__dict__['hybridization'] = np.array(hybridizations)
+        self.__dict__['neighbors'] = np.array(neighbors)
+        self.__dict__['bondtypes'] = np.array(bondtypes)
+        self.__dict__['_chiraltags'] = np.array(chiraltags)
+
+        if _mol.GetNumConformers() != 0:
+            coords = _mol.GetConformer(0).GetPositions()
+            self.__dict__['coords'] =  coords[:,:,np.newaxis]
+        else:
+            self.__dict__['coords'] = np.zeros((self.numAtoms,3,1))
 
-        # Add hydrogens
-        if addHs:
-            self._mol = Chem.RemoveHs(self._mol)
-            self._mol = Chem.AddHs(self._mol, addCoords=True)
+    def copy(self):
+        """
+        Create a copy of the molecule object
+
+        Returns
+        -------
+        newsmallmol : :class:`SmallMol`
+            A copy of the object
+        """
+        newsmallmol = deepcopy(self)
+        return newsmallmol
+
+    def setProp(self, key, value, overwrite=False):
+        """
+        Sets the property value based on the key
+
+        Parameters
+        ----------
+        key: str
+            The name of the property to set
+        value: int, str, float, object
+            The value of the property
+        overwrite: bool
+            If True, the property will be overwritten if already exist
+            Default: False
+
+        Example
+        -------
+        >>> sm.setProp('Ki', 25) # doctest: +SKIP
+        >>> sm.listProps
+        ['ligname', 'totalcharge', '_mol', Ki]
+        >>> sm.setProp('ligname', 'myMol', overwrite=True)
+        >>> sm.getProp('ligname')
+        'myMol'
+
+        """
+
+        if not isinstance(key, str):
+             raise ValueError('Wrong type {} for key.  Should be {} '.format(type(key), type('string')))
+
+        _props = self.listProps
+
+        if not overwrite and key in _props:
+            raise ValueError('The key passed {} already exists. Set "overwrite" as True to overwrite an existing key'.format(key))
+
+        self.mol_fields.append(key)
+        self.__dict__[key] = value
+
+    def setPropsAtoms(self, key, values, aIdxs=None):
+        """
+        Sets the property for one or more atoms
+
+        Parameters
+        ----------
+        key: str
+            The name of the property to set
+        values: list
+            The list of the values for the property
+        aIdxs: list
+            The list of the atom index to which set up the values
+            Default: None
+
+        Example
+        -------
+        >>> sm.setPropsAtoms('myTag', ['a', 'b', 'a', 'c'], aIdxs=[0,2,5,3]) # doctest: +SKIP
+        >>> sm.myTag
+        array(['a', None, 'b', 'c', None, 'a', None, None, None, None, None, None,
+       None, None, None, None, None, None], dtype=object)
+        """
+        class ValuesLengthErrors(Exception):
```

Comment:
rename to `ValueLengthError`. No plural needed if you see how `ValueError` is called.

Reply:
done it